types/mui-datatables/index.d.ts

```diff
@@ -159,120 +171,132 @@ export interface MUIDataTableIsRowCheck {
     ];
 }
 
-export interface MUIDataTableOptions {
-    caseSensitive?: boolean;
-    count?: number;
-    customFilterDialogFooter?: (filterList: any[]) => React.ReactNode;
-    customFooter?: (
+export type MUIDataTableOptions = Partial<{
+    caseSensitive: boolean;
+    confirmFilters: boolean;
+    count: number;
+    customFilterDialogFooter: (filterList: any[], applyNewFilters?: (...args: any[]) => any) => React.ReactNode;
+    customFooter: (
         rowCount: number,
         page: number,
         rowsPerPage: number,
         changeRowsPerPage: (page: string | number) => void,
         changePage: (newPage: number) => void
     ) => React.ReactNode;
-    customRowRender?: (data: any[], dataIndex: number, rowIndex: number) => React.ReactNode;
-    customSearch?: (searchQuery: string, currentRow: any[], columns: any[]) => boolean;
-    customSearchRender?: (
+    customRowRender: (data: any[], dataIndex: number, rowIndex: number) => React.ReactNode;
+    customSearch: (searchQuery: string, currentRow: any[], columns: any[]) => boolean;
+    customSearchRender: (
         searchText: string,
         handleSearch: any,
         hideSearch: any,
         options: any
     ) => React.Component | JSX.Element;
-    customSort?: (data: any[], colIndex: number, order: string) => any[];
-    customToolbar?: () => React.ReactNode;
-    customToolbarSelect?: (
-        selectedRows: {
-            data: Array<{ index: number; dataIndex: number }>;
-            lookup: { [key: number]: boolean };
-        },
+    customSort: (data: any[], colIndex: number, order: string) => any[];
+    customTableBodyFooterRender: (options: { data: any[]; selectableRows: SelectableRows; columns: any[] }) => any;
+    customToolbar: () => React.ReactNode;
+    customToolbarSelect: (
+        selectedRows: { data: Array<{ index: number; dataIndex: number }>; lookup: { [key: number]: boolean } },
         displayData: Array<{ data: any[]; dataIndex: number }>,
         setSelectedRows: (rows: number[]) => void
     ) => React.ReactNode;
-    disableToolbarSelect?: boolean;
```

Comment:
same topic of `@deprecated`:
https://github.com/gregnb/mui-datatables/blob/3.1.4/docs/v2_to_v3_guide.md#disabletoolbarselect-deprecated-in-favor-of-selecttoolbarplacement
> disableToolbarSelect will still work, but will give a deprecation warning